Code/src/main/java/nl/utwente/group10/ui/BackendUtils.java

```diff
@@ -31,4 +33,18 @@ public static Type dive(ConstT function, int steps) {
         }
         return function;
     }
+    
+    /**
+     * @return Whether or not t1 and t2 unify with each other.
```

Comment:
Its kind of impossible to display type mismatches if the front-end is not allowed to check for type mismatches.

If the typesMatch() method uses analyze thats fine with me.